WcaOnRails/spec/mailers/competitions_mailer_spec.rb

```diff
@@ -128,4 +128,36 @@
       expect(mail.body.encoded).to match(/Feb 1/)
     end
   end
+
+  describe "results_submitted" do
+    before :all do
+      @competition = FactoryBot.create(:competition, name: "Comp of the future 2017")
+
+      @results_file_path = "uploads/#{@competition.id}/results.json"
```

Comment:
We use `let`s to define variables instead of class attributes, you can see examples all around the spec files.

Reply:
Yes I have it here because I need the path in the teardown (to delete the file). 
But in this case I don't really need the file path to include the competition ID, I could just use the same results file every time.

Reply:
Oh, can you not access `let`s from within a `after :all` context?

One solution would be to use `Tempfile`, which looks like it automatically gets delete for you. See http://www.danielstjules.com/2013/09/11/testing-file-uploads-in-rails-without-fixtures/ for an example of it in action.

Reply:
I see you added a `let(:file)` below this. Do we still need this code and the `after :all` at the end?

Reply:
I'm cleaning this up. If passing the contents directly works then I won't have to use a file at all here!